backend/test/core/entity/CalculGroupBalance.test.ts

```diff
@@ -0,0 +1,261 @@
+          new Transaction(0, 0, 2),
+        ]);
+
+        // Act
+        const result = new CalculGroupBalance().calcul(group);
+
+        // Assert
+        const expectedResult = new GroupBalance(
+          1,
+          new Map<number, number>([
+            [0, 1],
+            [1, -1],
+          ])
+        );
+        expect(result).toEqual(expectedResult);
+      });
+      test("with payerId 0 and amount 3 for the whole group, it should return a balance {0 :3/2,1 :-3/2}", async () => {
+        // Arrange
+        const group = new Group(1, "group", members, [
+          new Transaction(0, 0, 3),
+        ]);
+
+        // Act
+        const result = new CalculGroupBalance().calcul(group);
+
+        // Assert
+        const expectedResult = new GroupBalance(
+          1,
+          new Map<number, number>([
+            [0, Number((3 / 2).toFixed(10))],
+            [1, Number((-3 / 2).toFixed(10))],
+          ])
+        );
+        expect(result).toEqual(expectedResult);
+      });
+    });
+
+    describe("Given a group of three members with one transaction", () => {
+      beforeEach(() => {
+        members = [
+          new GroupMember(0, "Luc"),
+          new GroupMember(1, "Jessica"),
+          new GroupMember(2, "Théo"),
+        ];
+      });
+      test("with payerId 0 and amount 0 for the whole group, it should return a balance null", async () => {
+        // Arrange
+        const group = new Group(1, "group", members, [
+          new Transaction(0, 0, 0),
+        ]);
+
+        // Act
+        const result = new CalculGroupBalance().calcul(group);
+
+        // Assert
+        const expectedResult = new GroupBalance(
+          1,
+          new Map<number, number>([
+            [0, 0],
+            [1, 0],
+            [2, 0],
+          ])
+        );
+        expect(result).toEqual(expectedResult);
+      });
+      test("with payerId 0 and amount 3 for the whole group, it should return a balance {0 :2,1 :-1, 2 :-1}", async () => {
+        // Arrange
+        const group = new Group(1, "group", members, [
+          new Transaction(0, 0, 3),
+        ]);
+
+        // Act
+        const result = new CalculGroupBalance().calcul(group);
+
+        // Assert
+        const expectedResult = new GroupBalance(
+          1,
+          new Map<number, number>([
+            [0, 2],
+            [1, -1],
+            [2, -1],
+          ])
+        );
+        expect(result).toEqual(expectedResult);
+      });
+      test("with payerId 0 and amount 4 for the whole group, it should return a balance {0 :8/3,1 :-4/3, 2 :-4/3}", async () => {
+        // Arrange
+        const group = new Group(1, "group", members, [
+          new Transaction(0, 0, 4),
+        ]);
+
+        // Act
+        const result = new CalculGroupBalance().calcul(group);
+
+        // Assert
+        const expectedResult = new GroupBalance(
+          1,
+          new Map<number, number>([
+            [0, Number((8 / 3).toFixed(10))],
```

Comment:
Did you have any issues with numbers that you used `toFixed(10)` ?

Reply:
If you try to compare fractions they don't always have the same amount of digit and it can bring the tests to fail. I arbitrary chose to keep only 10 digits in order to approximate the fraction.